docs/spec/index.md

```diff
@@ -102,11 +104,12 @@ In addition to the fields defined in GTFS, specific fields for use within TODS a
 
 ### `run_events.txt`
 
-Primary Key: (`service_id`, `run_id`, `event_sequence`)
+Primary Key: (`service_id`, `trip_service_id`, `run_id`, `event_sequence`)
 
 | **Field Name** | **Type** | **Required** | **Description** |
 | --- | --- | --- | --- |
-| `service_id` | ID referencing `calendar.service_id` or `calendar_dates.service_id` | Required | Identifies a set of dates when the run is scheduled to take place. |
+| `service_id` | ID referencing `calendar.service_id` or `calendar_dates.service_id` | Required | Identifies a set of dates when the run is scheduled to take place. Can be distinct from `trip_service_id`. |
```

Comment:
There should be some rules written down about what the rules are for how run service_ids can overlap with trip service_ids. Is it just that if a run works on a trip, the run's service_id can only happen on dates that the trip's service_id also happens? Or are there more complex rules about not having the same run happen twice on one date, or all the trips on one (service_id, run) pair all being on the same trip service_id?

Also, how would someone verify that a trip is always covered? That shouldn't be required in the spec (cuz the spec should allow producers to skip parts of the schedule) but I want to check that it'll possible for validators.

Reply:
Addressed via https://github.com/jeffkessler-keolis/operational-data-standard/pull/2

> If the run's `service_id` and the trip's `service_id` are different, the run's `service_id` must not occur on any dates that the trip's `service_id` doesn't (after combining data from `calendar.txt`, `calendar_supplement.txt`, `calendar_dates.txt`, and `calendar_dates_supplement.txt`).